fairscale/optim/oss.py

```diff
@@ -201,6 +202,72 @@ def step(self, closure: Optional[Callable[[], float]] = None, **kwargs: Any) ->
 
         return loss
 
+    def clip_grad_norm(self, max_norm: float, norm_type: float = 2.0) -> torch.Tensor:
+        """
+        Clip all gradients at this point in time. The norm is computed over all gradients together, as if they were
+        concatenated into a single vector. Gradients are modified in-place.
+
+        Arguments:
+            max_norm (float or int): max norm of the gradients
+            norm_type (float or int): type of the used p-norm. Can be ``'inf'`` for
+                infinity norm.
+
+        Returns:
+            Total norm of the parameters (viewed as a single vector).
+
+        .. note: This is analogous to `torch.nn.utils.clip_grad_norm_` but handles the partitioning and multiple devices per rank
+        under the hood.
+
+        .. warning: This needs to be called on all ranks, since synchronization primitives will be used
+        """
+
```

Comment:
As discussed over chat, I'll add in the model parallel case. In the meantime, could you add the below so that there aren't any silent bugs?

```suggestion
if self.group != dist.group.WORLD:
  raise NotImplemented("Clip norm not yet supported for model parallelism (coming soon!)")
```